tidb-lightning/tidb-lightning-error-resolution.md

```diff
@@ -0,0 +1,318 @@
+
+Unique/Primary key conflict in Local backend is handled separately, explained in the next section.
+
+## Duplicate resolution
+
+{{< copyable "" >}}
+
+```toml
+[tikv-importer]
+duplicate-resolution = 'record'
+```
+
+Local backend imports data by first converting them to KV pairs, and ingest them into TiKV in batches. Unlike the TiDB backend, duplicate rows are not detected until the end of task. Therefore, duplicate errors in local backend is not controlled by `max-error`, but rather a separate configuration `duplicate-resolution`.
+
+There are three possible values of `duplicate-resolution`:
+
+* **'none'** — Do not detect duplicates. If a unique/primary key conflict does exist, the imported table will have inconsistent data and index, and will fail checksum check.
+* **'record'** — Detect duplicates, but do not attempt to fix it. If a unique/primary key conflict does exist, the imported table will have inconsistent data and index, and will fail checksum check.
+* **'remove'** — Detect duplicates, and remove *all* duplicated rows. The imported table will be consistent, but the involved rows are ignored and have to be added back manually.
+
+TiDB Lightning duplicate resolution can only detect duplicates within the data source. It cannot handle conflict with existing data before running TiDB Lightning.
+
+## Error report
+
+{{< copyable "" >}}
+
+```toml
+[lightning]
+task-info-schema-name = 'lightning_task_info'
+```
+
+All errors are written to tables in the `lightning_task_info` database, in the downstream TiDB cluster. The database name can be changed with the configuration `lightning.task-info-schema-name`.
+
+TiDB Lightning creates 3 tables inside this database:
+
+```sql
+CREATE TABLE syntax_error_v1 (
+    task_id     bigint NOT NULL,
+    create_time datetime(6) NOT NULL DEFAULT now(6),
+    table_name  varchar(261) NOT NULL,
+    path        varchar(2048) NOT NULL,
+    offset      bigint NOT NULL,
+    error       text NOT NULL,
+    context     text
+);
+
+CREATE TABLE type_error_v1 (
+    task_id     bigint NOT NULL,
+    create_time datetime(6) NOT NULL DEFAULT now(6),
+    table_name  varchar(261) NOT NULL,
+    path        varchar(2048) NOT NULL,
+    offset      bigint NOT NULL,
+    error       text NOT NULL,
+    row_data    text NOT NULL
+);
+
+CREATE TABLE conflict_error_v1 (
+    task_id     bigint NOT NULL,
+    create_time datetime(6) NOT NULL DEFAULT now(6),
+    table_name  varchar(261) NOT NULL,
+    index_name  varchar(128) NOT NULL,
+    key_data    text NOT NULL,
+    row_data    text NOT NULL,
+    raw_key     mediumblob NOT NULL,
+    raw_value   mediumblob NOT NULL,
+    raw_handle  mediumblob NOT NULL,
+    raw_row     mediumblob NOT NULL,
+    KEY (task_id, table_name)
+);
+```
+
+**syntax_error_v1** is intended to record syntax error from files. It is not implemented yet.
+
+**type_error_v1** records all [data errors](#data-error) managed by the `max-error` configuration. There is one row per error.
```

Comment:
`data error` is a dead anchor. There is no `data errors` section in this file.

Reply:
```suggestion
**type_error_v1** records all [type errors](#type-error) managed by the `max-error` configuration. There is one row per error.
```